### ISSUE TITLE
Change color of a animation file

### ISSUE THREAD

**Issue by Pablouu** [airbnb/lottie-web#1666]
I want to change the color of this:
https://lottiefiles.com/35-loader

**Comment by Inventsable** [comment]
[Assign a class or ID](https://github.com/airbnb/lottie-web/wiki/SVG-and-HTML-ids-and-classes-to-reference-via-css-or-select-via-Javascript) to whatever fill is affecting your shapes:

&nbsp;&nbsp;&nbsp;&#x2023; &#x25a0; 1 &#x2605; Shape Layer 1<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&#x2023; Contents<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&#x2023; Ellipse Path 1<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&#x2023; Fill 1 <<< Rename this to a class, as in ".some-class"

Then reassign the color in CSS like so:

```css
.some-class {
  fill: red;
  stroke: red;
}
```

**Comment by weiesky** [comment]
have found：(Fill 1)
![image](https://user-images.githubusercontent.com/3365650/59177745-97826480-8b8f-11e9-8329-7c8992c834cf.png)
`
k: [
  0,
  0,
  0,
  1
]
`
which color do you prefer? 
Example:  #34f293
`
k: [
  0.203125, // 34 is 3x16 + 4 = 52, value = 52 / 256 = 0.203125
  0.9453125, // f2 is 15 x 16 + 2 = 242, value - 242 / 256 =  0.9453125
  0.57421875, // 93 ...
  1
]
`

**Comment by Pablouu** [comment]
I can't do it, i use a editor online but i don't understand the language, so i paste your code but nothing happens, i want to change the color to white

**Comment by weiesky** [comment]
https://assets4.lottiefiles.com/packages/lf20_6AbckL.json
White is ready for you.

```
k: [
  1,
  1,
  1,
  1
] // this is white, #FFFFFF
```

**Comment by vdias38** [comment]
Hi @weiesky I'm trying to change fill color using css class as you described above but it is not working. Can you help me please ?
I've created a [stackblitz](https://stackblitz.com/edit/mmb-lottie-performance?file=index.js), the [json file is a public gist](https://gist.githubusercontent.com/vdias38/29fb0766bc0d65c4e8a50d744b8a8f0b/raw/80c0a440c56c75c5413c4aeb810289727b93cbce/mmb-modern-4-slides.json) where I've renamed 2 layers '.square' (see screenshot below)

![https___gist_githubusercontent_com_vdias38_29fb0766bc0d65c4e8a50d744b8a8f0b_raw_80c0a440c56c75c5413c4aeb810289727b93cbce_mmb-modern-4-slides_json](https://user-images.githubusercontent.com/9011612/69055871-bae6f080-0a06-11ea-8426-e54fedf9be01.png)

And apply following css as you suggested

```css
.square {
  fill: green;
}
```

Do you see stg wrong ?

Thanks

**Comment by bodymovin** [comment]
hi @vdias38 instead of naming the layer can you try naming the fill properties with `.square`?

**Comment by vdias38** [comment]
Hi @bodymovin I've named ".square" the fill of shape layer, not the layer, sorry if It was not well expressed on my comment, I fix it.

**Comment by bodymovin** [comment]
@vdias38 Can you share the .aep project or a simplified link where to look for the class?

**Comment by vdias38** [comment]
Hi @bodymovin, sorry I haven't the aep, I'm manipulating directly the json, maybe it's why it not works... I've supposed renaming directly on json "Fill 1" should work do you confirm ?
I've isolated the json node path I've updated:
//"assets"/"id": "comp_9"/layers@"nm": "Shape Layer 1"/shapes@nm": "Rectangle 1"/it@nm": "Fill 1"
It concerns a shape rectangle 640x300.

I've renamed "Rectangle 1" to ".rectangle1", and "Fill 1" to ".fill1" but none of these classes appears on svg. Below the result on svg of json node highlighted above.
If I look for ".rectangle1" or ".fill1" on whole svg it returns none result. Then seems to have no effect to rename fill property on json, or **couldn't be done like this directly on json ?**

```xml
<g clip-path="url(#__lottie_element_196)" transform="matrix(0.0003433344536460936,0.9999999403953552,-0.9999999403953552,0.0003433344536460936,3658.390869140625,-373.87353515625)" opacity="1" style="display: block;">
    <g mask="url(#__lottie_element_223)" style="display: block;">
        <g transform="matrix(1.38100004196167,0,0,1.38100004196167,640,300)" opacity="1">
            <g opacity="1" transform="matrix(1,0,0,1,0,0)">
                <path fill="rgb(254,197,3)" fill-opacity="1" d=" M415,-150 C415,-150 415,150 415,150 C415,150 -415,150 -415,150 C-415,150 -415,-150 -415,-150 C-415,-150 415,-150 415,-150z"></path>
            </g>
        </g>
    </g>
</g>
```

**Comment by bodymovin** [comment]
I see, those are the property names, but not the classNames, you will need to add an additional property "cl" to them with the class value for it to work.

**Comment by vdias38** [comment]
@bodymovin that's the solution!! I've tested with `cl` and it works fine, thanks a lot for your support.

**Comment by collinjoyce** [comment]
In case anyone's still watching this or comes across it like I did, https://edit.lottiefiles.com/ saved my ass on an identical request.

**Comment by MamaMightyMac** [comment]
@bodymovin Hi! I have been trying to understand this thread because I have another designer that needs to use an animation I created on a whole group of client sites that all have different color schemes. The hope was that I could create one animation and let the stylesheets for each site dictate color, stroke widths...I was following along until you got the adding "cl" and then magic happens and everyone's ok but I'm lost how to do that. I am NOT a developer. (I'm rusty on my html/css ) Please explain in painful, newbee style detail how I can do this. I'm also trying to champion Lottie for all our applications and products so finding ways to customize without creating assets every single time would be great. (like if bodymovin' could capture the master properties so that I could alternately create multiple color versions in the precomps instead of just capturing the master file's attributes.) Thank you.  :-)

**Comment by Inventsable** [comment]
> [Assign a class or ID](https://github.com/airbnb/lottie-web/wiki/SVG-and-HTML-ids-and-classes-to-reference-via-css-or-select-via-Javascript) to whatever fill is affecting your shapes:
> 
>    ‣ ■ 1 ★ Shape Layer 1
>          ‣ Contents
>                ‣ Ellipse Path 1
>                ‣ Fill 1 <<< Rename this to a class, as in ".some-class"
> 
> Then reassign the color in CSS like so:
> 
> ```css
> .some-class {
>   fill: red;
>   stroke: red;
> }
> ```

@MamaMightyMac, if you have control of the AEP file then it honestly is this simple, and you can just reassign the particular class attributes via a stylesheet. People are wanting to do it programmatically in JS in this thread, but that's not a very effective solution in my opinion and completely unnecessary now that Lottie has two-way compatibility and you can import into AE.

**Comment by MamaMightyMac** [comment]
Hi,

Thank you so much for responding!!!!!!!!

I cannot figure out HOW to change the Fill 1 property name to a Class name except by searching effects. Here is a screenshot.  I’ve Gone to the link you provided too. I can easily change the layer name but not the Fill or Stroke. (And I cannot find any way at all for stroke.) I’ve been on this for days….

**Comment by Inventsable** [comment]
Hi, I don't see a screenshot. Are you working with a locked file or did you make this yourself? You just change the layer or property name, as in `Fill` (the property group, not property itself. The one with the twirldown). If you want to make it a class and it's currently `Fill 1`, you can rename it to `.some-class`. Lottie recognizes any property name or layer prefixed with `.` as a class you can use with CSS, and `#` as an id. You rename it the same way as any thing else in After Effects, double click or select and hit Enter. Nothing to do with Effects panel:

![ice_screenshot_20200615-174756](https://user-images.githubusercontent.com/37279677/84719533-738af900-af30-11ea-9b60-710247b8b2da.png)

**Comment by MamaMightyMac** [comment]
That’s the thing…. I cannot for the life of me, change the name from Fill 1 to anything else, same with Stroke.  I created the file myself. I’ve attached the AE.  This seems so easy but by golly - nothing.  Could it be a Mac vs Windows thing?

**Comment by Inventsable** [comment]
Curious, are you just learning AE, or is this a recent change due to an update? Do you see error or bug reports about it?

**Comment by MamaMightyMac** [comment]
Yes, I started learning AE about a year ago, specifically for Lottie. My company wants to add animations to their products but can’t settle on anything since we always have to have parity btwn iOS, Android & web. 
So I apologize if I don’t know all the cool shortcuts. But gosh it shouldn’t be this hard when I can change the layer names with ease.

I created two animations - one for when a product is activating - so the amount of time it could take varies (Clock) and when it is complete - Check.  So then I started trying to figure out how to change things on the fly but not using the LottieFiles web-based tool that creates new assets.  Ideally - one asset controlled by stylesheets. Perfect.  

I”ve been trying to edit the demo.html file in dreamweaver as well , to add the class name in the style and change the color of say, the background of the check, but nothing works.

**Comment by Inventsable** [comment]
> Ideally - one asset controlled by stylesheets. Perfect.

This is how the original Bodymovin panel, not the LottieFiles one, works if you rename and assign classes/ids then use CSS to alter the values. Just to be clear though, the following does not work:

- Create a new shape layer
- Twirl the layer open, click Add > on the Contents line.
- Choose fill
- Twirl open to see Fill 1 group, select it and only it (may need to click empty space in Comp viewer to deselect everything else).
- Press Enter

You can else Edit > Keyboard Shortcuts and search for Rename, which has the same effect

**Comment by MamaMightyMac** [comment]
Press Enter
Oh. My.  Goodness. 

Amazing.  Appreciating the simplest of things in life right now. (And feeling stupid) 

So renaming only works for Bodymovin’ output, not LottieFiles.  I can live with that. I will try again. 

I can see it now in the json file. :-) Now I just have to figure out how to make it appear in an html page.

**Comment by Inventsable** [comment]
We've all been there. Also just to clarify, I don't know that LottieFiles works the exact same way with their format so I can only guarantee this method will work with Bodymovin' but am unsure about how LottieFiles handles it. Don't actually know that it isn't supported with LottieFiles in case that route would be easier for you.

**Comment by MamaMightyMac** [comment]
Thank you again SO MUCH.  TWO DAYS to find out I just press “enter.”  Oh man, that is one for the books.

Virtual hug. 

Now I can go make dinner.

**Comment by xxmuaddib** [comment]
You can also use my library to change colors of a lottie animation from code - https://github.com/xxmuaddib/lottie-colorify

**Comment by AhyaH-YHWH** [comment]
Then FF becomes 15*16 +15=256
and FE becomes 15*16+ 14= 255
256/255 is greater than 1
#FED60C
#FFD60C
We should use same Number 1 of both FE, FF?

D6=13*16+6=214
0C=0*16+12=12

**Comment by techwil** [comment]
@xxmuaddib it's not working at my end i tried it so..

**Comment by xxmuaddib** [comment]
@techwil Can you post an example so I could help you? As it is another project, I would suggest posting the example there.

**Comment by techwil** [comment]
The example isn't working.

**Comment by xxmuaddib** [comment]
You can pull it again, it will work now. `lib` folder wasn't included in the repo. You could build it using `yarn build`. But now I have added it to the repo. However, IMHO it is inappropriate answering questions about other package here. So if you will need further assistance, ask questions in `lottie-colorify` repo.

**Comment by techwil** [comment]
@bodymovin is there any way to manipulate the Lottie json file with a new color?

**Comment by spoeken** [comment]
How about setting the color to `currentColor`, how would that be done?

**Comment by techwil** [comment]
@spoeken I'm working on something big would You like to be a part of it?

**Comment by smostafad** [comment]
Hi guys, I want to have an icon editor similar to www.lordicon.com, but when I edit the json file on the site by inspect, the color changes, but whatever I do with the coding, it does not change anything Can you help me ?.

**Comment by Inventsable** [comment]
> hi @vdias38 instead of naming the layer can you try naming the fill properties with `.square`?

Funny that every one gives me a thumbs down for saying exactly the same thing as the creator of Lottie, lol. Unless you have a very rare edge case where you need to use JS, you should be using CSS to do this. Giving valid answers thumbs down isn't good Github etiquette -- just because it doesn't solve your problem doesn't mean it won't solve someone else's.

> Hi guys, I want to have an icon editor similar to [www.lordicon.com](http://www.lordicon.com), but when I edit the json file on the site by inspect, the color changes, but whatever I do with the coding, it does not change anything Can you help me ?.

You'd have to edit the JSON before loading your animation into lottie. Are you? No changes to the JSON after loading it would have any effect on the animation. I'd still recommend doing it by CSS even when programmatically -- traverse through all shapes and when a `cl` key/value isn't found, create one for that shape, then load the animation and use CSS to control the color so it can be modified at any time after load and during playback.

**Comment by smostafad** [comment]
In After Effects, I gave it an id and  change it with css code, but as a color picker, every Ready codes that I put does not change anything. Do you have a code that can help me?

**Comment by Inventsable** [comment]
Sorry, I don't understand what you mean by that. What does "as a color picker, every Ready codes that I put does not change anything" mean?

**Comment by Inventsable** [comment]
If you're wanting to have part of a lottie animation that changes according to color picker input, I'd assign a `cl` (or do it in AEP as suggested initially) to the shape's fill you want to target. Use CSS variables inside that fill, and reassign the value of the variable to the color picker's result.

**Comment by smostafad** [comment]
I mean, I put the id in After Effects on a JavaScript id to change the color, but it does not change at all.

**Comment by Inventsable** [comment]
Can you provide an example? Screenshot of AEP, JSON, and/or CSS being used?

**Comment by smostafad** [comment]
For example, after a effect, I consider a name layer to give as an id in JavaScript so that the color changes when the color pick is opened.

**Comment by Inventsable** [comment]
Your phrasing of things is very hard to understand. Can you give screenshots or code instead?

**Comment by Inventsable** [comment]
Ideally:

- Screenshot of the ID inside After Effects
- Link to lottie JSON
- CSS selector or JS snippet where you use the id but it isn't working

**Comment by smostafad** [comment]
![Screenshot (76)](https://user-images.githubusercontent.com/90977545/133906239-c3cd3372-e3fb-41a3-a9dc-022f7b01071f.png)
![Screenshot (75)](https://user-images.githubusercontent.com/90977545/133906242-ccbf8bf4-e34e-435e-85aa-9fb152aa8e13.png)
 The id I gave in After Effects works on the site, but when I call in JavaScript, its color does not change at all.

**Comment by Inventsable** [comment]
All correct so far.. How are you calling and changing the color in JS?

**Comment by smostafad** [comment]
![Screenshot (77)](https://user-images.githubusercontent.com/90977545/133906364-6e9fe7cf-3691-4060-ad14-1e2fb247ba5d.png)

**Comment by Inventsable** [comment]
#color is an SVG element, not an HTML one:

```css
#color {
    fill: black;
}
```

You need to use the `fill` attribute of CSS instead of treating it like an HTML element and trying `background`/`background-color`.

**Comment by smostafad** [comment]
#color{
stroke : red !important 
}

It changes with this, but I want it to be a color pick

**Comment by Inventsable** [comment]
I'd personally use CSS variables instead, altogether:

```css
:root {
  --someColor: #ff0000;
}

#color {
    fill: var(--someColor);
}
```

Then in JS:

```js
function setCSS(key, value) {
  document.documentElement.style.setProperty('--' + prop, data);
}

// Whenever we need to change it:
setCSS("someColor", "#009900") // can use any valid CSS color value (hex, HSL, RGB, RGBA, named color, etc)
```

**Comment by smostafad** [comment]
![Screenshot (80)](https://user-images.githubusercontent.com/90977545/133906506-7c5ba46e-bbc5-4839-8421-d662f9394a4d.png)
With this change, its color turned black, but I will change its color picker. Nothing happens. Do you have a solution for that?

**Comment by Inventsable** [comment]
See above comment about CSS variables 👍

**Comment by Inventsable** [comment]
Your color picker also needs to not target `style.backgroundColor` on value change. This is one of the many reasons why CSS is far easier and more efficient to use for styling than JS as I've maintained throughout the thread, ha. You'd need to target `fill` or ideally use CSS variables on the color picker callback.

**Comment by smostafad** [comment]
I do not know much JavaScript that I put the code correctly?

**Comment by Inventsable** [comment]
Copy and paste your HTML here so I don't have to manually type it out and I'll put it in for you.

**Comment by Inventsable** [comment]
```html
<html>
  <head>
    <style>
      :root {
        --sample: black;
      }
      #color {
        fill: var(--sample);
        width: 500px;
        height: 100px;
      }
    </style>
  </head>

  <body>
    <div class="color" onclick="changeColor"></div>
    <input name="MyColorPicker" type="color" id="MyColorPicker" />
    <script>
      function setCSS(key, value) {
        document.documentElement.style.setProperty(`--${key}`, value);
      }
      function changeColor() {
        let newValue = document.getElementById("MyColorPicker").value; // Get the color picker value
        setCSS("sample", newValue); // Reassign our CSS var with it
      }
    </script>
  </body>
</html>
```

**Comment by Inventsable** [comment]
Guess the problem is solved then?

**Comment by smostafad** [comment]
Thank you for that, but unfortunately it did not change again

**Comment by smostafad** [comment]
I have to do something special inside the json file?

**Comment by Inventsable** [comment]
My code didn't include the lottie invocation, neither does yours. Are you loading lottie and the JSON into HTML still? The code I provided above does not handle or include that section.

**Comment by Inventsable** [comment]
The code you showed me has a `div#color`, which is not Lottie or your JSON. It's just a placeholder HTML element. I'm assuming you use that as your `container` to load Lottie and have a separate file or code somewhere -- unless this is literally the only code you have, in which case you're not using Lottie at all just yet. The color picker won't work on my example because `background-color` works on HTML whereas `fill` works on SVG. If you were to change my example to have `background-color` as you did originally, it should.

You aren't loading a lottie animation anywhere here, though.

**Comment by smostafad** [comment]
Yes, I deleted my code and left yours, but when I open the color picker, the color of the animation does not change.

**Comment by Inventsable** [comment]
What is your lottie code? If `div#color` is your container, that means you have two elements with an id of `#color`. One `div#color`, and another `path#color` (which is Lottie).

**Comment by smostafad** [comment]
path#color

**Comment by Inventsable** [comment]
`path#color` and `div#color` are both queried by `document.getElementById("color")`. You cannot have duplicate ids. By "lottie code" I mean where in your code are you loading Lottie as a CDN, invoking it and passing your JSON into?

**Comment by Inventsable** [comment]
Honestly, can you just make a new issue? This is getting a bit out of hand.

**Comment by smostafad** [comment]
I will send the file to you to see where the problem is?

**Comment by smostafad** [comment]
[test.zip](https://github.com/airbnb/lottie-web/files/7190573/test.zip)
It does not have an animation just to test it and it has a #color id

**Comment by Inventsable** [comment]
I'd rather you create a new issue at this point to be honest, this is getting out of scope of relating to the original issue. Host a repo on Github with everything included and I'll look at it eventually, but I'm running out of free time on my end.

**Comment by Inventsable** [comment]
Yes that's the lottie JSON, but that's **not the code in HTML or Javascript that you're loading and invoking Lottie with**, lol. Please provide the code you're invoking and loading Lottie with, where this JSON is being used in HTML.

**Comment by smostafad** [comment]
I use the elementor, it has an element called lottie, I uploaded it there and added it in an html

**Comment by smostafad** [comment]
I just turned the fill part into a stroke and it turned black and red, but it doesn't work inside the color picker. Could this be the reason?

**Comment by Inventsable** [comment]
I don't know what that is so it's hard to say if that contributes to the problem. You should change the `div#color` to a different id (not shared by Lottie, you cannot have duplicate ids) and verify the color picker's `.value` attribute provides a valid CSS color format -- if it returns "ff0000" instead of "#ff0000" this would be a good reason the example didn't work (along with duplicate ids being used, you cannot use duplicate ids).

**Comment by smostafad** [comment]
Thank you so much for your help

**Comment by Inventsable** [comment]
```html
<html>
  <head>
    <style>
      :root {
        --sample: black;
      }
      /* DO NOT USE DUPLICATE IDs */
      #container {
        width: 500px;
        height: 100px;
      }
      /* This is for SVG path ONLY */
      #color {
        fill: var(--sample);
      }
    </style>
  </head>

  <body>
    <!-- DO NOT USE DUPLICATE IDs, use something distinct like "container" -->
    <div id="container" onclick="changeColor" />
    <!-- Above you have it only changing color when clicked. Is this intentional? -->

    <input name="MyColorPicker" type="color" id="MyColorPicker" />
    <script>
      // Are you trying to change the color as the picker changes? If so, you'd need a listener:
      document.getElementById("MyColorPicker").addEventListener(
        "input",
        (evt) => {
          setCSS("sample", evt.target.value);
        },
        false
      );
      function setCSS(key, value) {
        document.documentElement.style.setProperty(`--${key}`, value);
      }
    </script>
  </body>
</html>

```

**Comment by smostafad** [comment]
broooooo change it

**Comment by smostafad** [comment]
Thank maaan

**Comment by smostafad** [comment]
What did you change?

**Comment by Inventsable** [comment]
You never had a listener on your color picker. You had `changeColor` as an `onclick` function only being triggered when you clicked the container for Lottie, which has nothing to do with the color picker. You also had duplicate id's, which is a problem when you query it.

Some advice: go invest in some more basic HTML, CSS and JS classes. If you spend a week taking classes, you'll solve your future problems in a few minutes instead of a few hours. The return on investment is well worth it. Don't try to play Beethoven before Mary Had a Little Lamb. Learn more about basic Javascript since the process of working with Lottie is going to be 100x more difficult when you don't know much about it than it would be otherwise. There's a million free resources for learning more about web dev out there -- it's worth doing. You set yourself up for failure and frustration when you jump into things and try to figure them out as you go. Find a teacher instead and aim for things just above your skill level instead of being too far above your current ability.

**Comment by smostafad** [comment]
And another question when the color of the animation changes and the file that is on the site host also changes for download

**Comment by Inventsable** [comment]
> And another question when the color of the animation changes and the file that is on the site host also changes for download

I don't know what this means. You're going to have to phrase your questions differently or explain more, I'm not trying to offend you but I can't understand what you mean when you ask questions half the time. Please take a minute to structure a question or provide screenshots instead of typing so quickly otherwise we have to go back and forth for a few replies before I can get what you're trying to ask.

**Comment by smostafad** [comment]
Excuse me for that, I mean, for example, the site www.lordicon.com, when you change the color of your animation and download the json file, it is exactly the same color that I changed. I wanted to know if you have a solution for this, that is, when you load the json file, And do you allow it to be downloaded, is it possible for the user to change its color, and I can download the same?

**Comment by Inventsable** [comment]
Well... The JSON file is static and has static values. The HTML rendered version of your Lottie animation (which originally started from the JSON data) is dynamic and can be changed but has no relation to the JSON after you load that JSON to start it. If you have web elements like a color picker that changes the value, it will only affect the current and present rendered version in your own HTML. If you want to change a color and then **make this a permanent change to the JSON so that you can load the new color in animation somewhere else**, you'd have to traverse through the JSON, find the corresponding shape (either by id or class) and use Javascript to parse the color manually. Consider the original example in this thread above:

```json
k: [ 0.2039215686, 0.9480196078, 0.5764705882, 1]
```

This is an Array of RGBA values: Red, Green, Blue, and Alpha (opacity/transparency). Unlike most RGB formats, AE uses a range of 0 - 1 instead of 0 - 255. A big factor in learning how to code is breaking things down into logical steps that you can articulate, then going through them from top to bottom and replacing the idea with tangible code. I'll break down the steps for you, but not outright give you the solution (plenty of help can be found on Googling it though):

- When the color picker for "#color" changes, store the value as a variable
- Iterate through the JSON's `layers` and each layer's `shapes` to find an entry with an `id` of `color`
- Once we find the correct `shape`, we can access the original color as `shapes.c.k`. This is our RGBA array. We need to rewrite this part of the JSON and save it
- Take our hexadecimal color string and break it down to an RGB equivalent of 0 - 255 for each (i.e. "#ff0000" should become `[255, 0, 0]`) - plenty of Stack Overflow threads on how to do this
- We need this as a range of 0 - 1, not 0 - 255. So we transpose each number in the RGB section to be 0 - 100, and when done, divide each number by 100.
- We can assume the `A` in `RGBA` is always `1`, unless there's an opacity effect. Since opacity is already 0 - 100, we just divide opacity by 100 if it exists (otherwise it's value would already be 100)
- We reassign `shape.c.k` to be equal to our new value
- The user can now download the animation with the new color, problem solved

---

This isn't an easy solution or problem to solve. It could take a lot of Javascript to do or very little but that depends on your knowledge of JS. Each step above is something you can research independently though, so just start at the beginning of the list and work your way down -- but don't skip parts. You have to do them in sequence, though it's okay if it's a bit sloppy on your first few tries.

**Comment by smostafad** [comment]
I think you mean to use the JavaScript code on the site when the user changes the color to change the json file dynamically?

**Comment by smostafad** [comment]
![Screenshot (84)](https://user-images.githubusercontent.com/90977545/133923964-2a883e4e-23c5-40f3-a1f0-0e5fe5c5f5f1.png)
![Screenshot (85)](https://user-images.githubusercontent.com/90977545/133923971-f9d208cd-a13b-4385-9790-aa0f48f2d2ba.png)

I search for the points you said and found two codes. Do you think these work?

**Comment by xxoutdated** [comment]
Still facing issue with coloring the animation ,
So basically I have a weird website so basically it changes color on reload and its done through jQuery I was wondering if there is a way to change the color of the path with jQuery I did try adding a class using normal css but can do it please help, TY in advance

**Comment by ismaelbarry** [comment]
I may be late here, but Lottie introduced their editor: https://lottiefiles.com/blog/updates/introducing-the-lottie-editor

With the editor, you can drag and drop your Lottie file and edit each layer (e.g., color, opacity, etc.) separately. I actually had the same issue where I downloaded an animation which wasn't supposed to have any red in it. However, after rendering it, the red appeared. By using the editor, I was able to find the layer that was indeed red and change it to the color and opacity that I needed.

**Comment by wibawasuyadnya** [comment]
> https://assets4.lottiefiles.com/packages/lf20_6AbckL.json White is ready for you.
> 
> ```
> k: [
>   1,
>   1,
>   1,
>   1
> ] // this is white, #FFFFFF
> ```

thank you pall, thank you so muchh

**Comment by noethi** [comment]
@Inventsable 
Thank you so much for your help! Your idea all works quite well, it's just that on the touch devices the fill level selection doesn't work and the lottie is displayed in black by default.

**Comment by EdDaWord** [comment]
### Web with JS
If you're looking for a javascript way of doing things it might be best to use this library and making use of `Flatten`: 
https://github.com/xxmuaddib/lottie-colorify alongside the lottie-web player

### Android & iOS
If you're looking for an iOS/Android way of doing things there are ways to use `setValueProvider`: https://swiftsenpai.com/development/lottie-value-providers/

### Editing the Lottie File's `k: [ r, g, b, a]` value
Like the comments listed above you can edit the Lottie file using an editor 
https://edit.lottiefiles.com/

### Editing the 
You can add a class name to the Lottie JSON in the form of `nm` and have a class in your CSS modify that color.

**Comment by silllli** [comment]
If your goal is to replace certain colors on the web, I found a simple solution: using `fill` and `stroke` values as CSS selector.

So if you have a `path` with a black fill, you can change its color to red this way:
```css
path[fill="rgb(0,0,0)"] {
  fill: rgb(255,0,0);
}
```

Or to use it with a more flexible `currentColor`:

```css
path[fill="rgb(0,0,0)"] {
  fill: currentColor;
}
```

To address multiple different origin and target colors, you could set the elements’ colors to unique placeholder RGB values—then use these RGB values as selector per desired “color group”.

**Comment by khaledOghli** [comment]
there is an easy way 
```

// js
 anim = lottie.loadAnimation({
        container: lavContainer.value,
        renderer: 'svg',
        animationData: json,
        autoplay: false,
        loop,
        rendererSettings: {
          className: 'lottie_svg',
        },
});
      
// css
.lottie_svg path[fill] {
  fill:currentColor !important;
}
.lottie_svg path[stroke] {
  stroke:currentColor !important;
}

// component pass the color
        <IconsGlobe
          ref="iconRef"
          class="h-7 w-7 text-primary-500"
        />



```
